packages/grid/_modules_/grid/GridComponent.tsx

```diff
@@ -3,180 +3,36 @@
-    useRenderInfoLog(apiRef, logger);
 
-    const showNoRowsOverlay = !props.loading && totalRowCount === 0;
-    const showNoResultsOverlay = !props.loading && totalRowCount > 0 && visibleRowCount === 0;
+    useGridComponent(apiRef, props);
 
-    const ariaProps = {
-      'aria-label': props['aria-label'],
-      'aria-labelledby': props['aria-labelledby'],
-    };
     return (
-      <GridApiContext.Provider value={apiRef}>
-        <NoSsr>
-          <GridRoot
-            ref={handleRef}
-            className={clsx(internalOptions.classes?.root, props.className)}
-            {...ariaProps}
-          >
-            <ErrorBoundary
-              hasError={errorState != null}
-              componentProps={errorState}
-              api={apiRef!}
-              logger={logger}
-              render={(errorProps) => (
-                <GridMainContainer>
-                  <components.ErrorOverlay
-                    {...errorProps}
-                    {...props.componentsProps?.errorOverlay}
-                  />
-                </GridMainContainer>
-              )}
-            >
-              <div ref={headerRef}>
-                <components.Header {...props.componentsProps?.header} />
-              </div>
-              <GridMainContainer>
-                <Watermark licenseStatus={props.licenseStatus} />
-                <GridColumnsContainer ref={columnsContainerRef}>
-                  <GridColumnsHeader ref={columnsHeaderRef} />
-                </GridColumnsContainer>
-                {showNoRowsOverlay && (
-                  <components.NoRowsOverlay {...props.componentsProps?.noRowsOverlay} />
-                )}
-                {showNoResultsOverlay && (
-                  <components.NoResultsOverlay {...props.componentsProps?.noResultsOverlay} />
-                )}
-                {props.loading && (
-                  <components.LoadingOverlay {...props.componentsProps?.loadingOverlay} />
-                )}
-                <GridAutoSizer
-                  onResize={onResize}
-                  nonce={props.nonce}
-                  disableHeight={props.autoHeight}
-                >
-                  {(size: any) => (
-                    <GridWindow ref={windowRef} size={size}>
-                      <GridViewport ref={renderingZoneRef} />
-                    </GridWindow>
-                  )}
-                </GridAutoSizer>
-              </GridMainContainer>
-              {!internalOptions.hideFooter && (
-                <div ref={footerRef}>
-                  <components.Footer {...props.componentsProps?.footer} />
-                </div>
-              )}
-            </ErrorBoundary>
-          </GridRoot>
-        </NoSsr>
-      </GridApiContext.Provider>
+      <GridContextProvider apiRef={apiRef} props={props}>
+        <GridRoot ref={ref}>
+          <GridErrorHandler>
+            <GridHeaderPlaceholder />
+            <GridBody />
+            <GridFooterPlaceholder />
```

Comment:
The terminology of "Placeholder" might not be correct, as it has a guest by default; Maybe:

```suggestion
            <GridHeaderContainer />
            <GridBody />
            <GridFooterContainer />
```